### ISSUE TITLE
Add @property to the no-types rules

### ISSUE THREAD

**Issue by profaurore** [gajus/eslint-plugin-jsdoc#1376]
## Motivation

Being able to disallow types on `@property` descriptions in Typescript files.

## Current behavior

The `no-types` rule currently only covers the `@param` and `@returns` descriptions.

## Desired behavior

The `no-types` rule should include `@property` descriptions.

## Alternatives considered

The `no-types` rule should have a setting to optionally include `@property` descriptions.

**Comment by profaurore** [comment]
Apologies for not responding to the first question. It is not clear to me, at least from reading the documentation, _why_ the default context is limited to functions, so it is difficult for me to answer.

I would proppose that the rule applies to the `ClassDeclaration` context, where explicit type definitions may conflict with Typescript definitions.
```ts
  /**
   * @class
   * @property {object} x
   */
class X {
  x: number;
}
```

**Comment by brettz9** [comment]
Thanks for the report. I've applied when found on a `ClassDeclaration` (a context which is now added to the checks). We don't want to be overly aggressive in the contexts as there could be `@typedef`s with `@property` tags whose types are needed.
